go/p2p/api/api.go

```diff
@@ -83,8 +83,16 @@ type Service interface {
 	RegisterProtocol(p core.ProtocolID, minPeers int, totalPeers int)
 
 	// RegisterProtocolServer registers a protocol server for the given protocol.
+	//
+	// Normally, registered protocol server should be also advertised (AdvertiseProtocol),
+	// so that remote peers find the host node.
 	RegisterProtocolServer(srv rpc.Server)
 
+	// AdvertiseProtocol starts advertising the host as available for serving the specified protocol.
+	//
+	// It should be called only after the protocol server is running and ready to serve.
```

Comment:
Is this true? I think you can call this method anytime, as it will only add protocol to the map of advertised namespaces, and when the discovery starts, the advertising begins.